sklearn/model_selection/_validation.py

```diff
@@ -801,21 +862,22 @@ def learning_curve(estimator, X, y, groups=None,
         out = parallel(delayed(_incremental_fit_estimator)(
             clone(estimator), X, y, classes, train, test, train_sizes_abs,
             scorer, verbose) for train, test in cv_iter)
+        out = np.asarray(out).transpose((2, 1, 0))
     else:
         train_test_proportions = []
         for train, test in cv_iter:
             for n_train_samples in train_sizes_abs:
                 train_test_proportions.append((train[:n_train_samples], test))
 
         out = parallel(delayed(_fit_and_score)(
-            clone(estimator), X, y, scorer, train, test,
+            clone(estimator), X, y, {'score': scorer}, train, test,
```

Comment:
I think ``_fit_and_score`` should take care of that and you shouldn't need to worry about that here...